File: src/main/scala/Extensions.scala

extension (s: String)
  def capitalizeFirstLetter: String = if s.isEmpty then s else s.charAt(0).toUpper + s.substring(1)

@main def extensions(): Unit = {
  val input = "hello, world!"
  val output = input.capitalizeFirstLetter

  println(output) // Output: Hello, world!
}

File: src/main/kotlin/Extensions.kt
package com.eventloopsoftware

fun String.capitalizeFirstLetter(): String =
    if (this.isEmpty()) this else this.substring(0, 1).uppercase() + this.substring(1)

fun main() {
    val input = "hello, world!"
    val output = input.capitalizeFirstLetter()

    println(output) // Output: Hello, world!
}